### PR TITLE
Only registered discovery integrations are eligible Adobe Vendor ID services.

### PR DESCRIPTION
**Pull request by tdilauro**
## Description

- Allows only discovery integrations with which a library has successfully registered to 
- Refactors some registration-related constants to avoid circular dependencies.
- Fixes test setup to set the registration status to successful for existing tests, so that they work as expected.

NB: This couples AuthdataUtility to registration more than I would like, but this is effectively the case with the currently available integrations.

## Motivation and Context

The exceptions generated by what looked like improperly configured vendor ID integrations are causing long response delays immediately after web server restarts.

## How Has This Been Tested?

- Added new test to ensure that unsuccessful registrations don't provide an AuthdataUtility.
- Manually tested to verify that the messages no longer appear in the logs.

## Checklist:

- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

File: api/adobe_vendor_id.py [hunk 1]
```diff
@@ -727,7 +728,10 @@ def from_config(cls, library, _db=None):
             vendor_id = ConfigurationSetting.for_externalintegration(
                 cls.VENDOR_ID_KEY, possible_integration
             ).value
-            if vendor_id:
+            registration_status = ConfigurationSetting.for_library_and_externalintegration(
+                _db, RegistrationConstants.LIBRARY_REGISTRATION_STATUS, library, possible_integration
+            ).value
+            if vendor_id and registration_status == RegistrationConstants.SUCCESS_STATUS:
```
Comment (tdilauro) [inline api/adobe_vendor_id.py:734]: This is the crux of the change.

File: api/testing.py [hunk 2]
```diff
@@ -81,6 +82,9 @@ def initialize_adobe(self, vendor_id_library, short_token_libraries=[]):
             ConfigurationSetting.for_library_and_externalintegration(
                 self._db, ExternalIntegration.PASSWORD, library, self.registry
             ).value = secret
+            ConfigurationSetting.for_library_and_externalintegration(
+                self._db, RegistrationConstants.LIBRARY_REGISTRATION_STATUS, library, self.registry
+            ).value = RegistrationConstants.SUCCESS_STATUS
```
Comment (tdilauro) [inline api/testing.py:87]: Existing tests were expecting a valid registration, so here we set the needed status.

File: tests/test_adobe_vendor_id.py [hunk 3]
```diff
@@ -624,6 +625,33 @@ def setup_method(self):
             },
         )
 
+    @pytest.mark.parametrize(
+        "registration_status, authdata_utility_type",
+        [
+            (RegistrationConstants.SUCCESS_STATUS, AuthdataUtility),
+            (RegistrationConstants.FAILURE_STATUS, type(None)),
+            (None, type(None)),
+        ]
```
Comment (tdilauro) [inline tests/test_adobe_vendor_id.py:634]: New test to verify that we don't get an AuthdataUtility unless a successful registration has been accomplished.

File: api/adobe_vendor_id.py [hunk 4]
```diff
@@ -722,22 +727,31 @@ def from_config(cls, library, _db=None):
...
+                    _db,
+                    RegistrationConstants.LIBRARY_REGISTRATION_STATUS,
+                    library,
+                    possible_integration,
+                ).value
+            )
+            if (
+                vendor_id
+                and registration_status == RegistrationConstants.SUCCESS_STATUS
+            ):
```
Comment (tdilauro) [inline api/adobe_vendor_id.py:745]: This is the crux of the change.

File: api/testing.py [hunk 5]
```diff
@@ -81,6 +82,12 @@ def initialize_adobe(self, vendor_id_library, short_token_libraries=[]):
             ConfigurationSetting.for_library_and_externalintegration(
                 self._db, ExternalIntegration.PASSWORD, library, self.registry
             ).value = secret
+            ConfigurationSetting.for_library_and_externalintegration(
+                self._db,
+                RegistrationConstants.LIBRARY_REGISTRATION_STATUS,
+                library,
+                self.registry,
+            ).value = RegistrationConstants.SUCCESS_STATUS
```
Comment (tdilauro) [inline api/testing.py:90]: Existing tests were expecting a valid registration, so here we set the needed status.

File: tests/test_controller.py [hunk 6]
```diff
@@ -542,6 +541,13 @@ def test_exception_during_short_client_token_initialization_is_stored(self):
         )
         registry_integration.username = "something"
         registry_integration.set_setting(AuthdataUtility.VENDOR_ID_KEY, "vendorid")
+        # Indicate that the library has successfully registered.
+        ConfigurationSetting.for_library_and_externalintegration(
+            self._db,
+            RegistrationConstants.LIBRARY_REGISTRATION_STATUS,
+            self.library,
+            registry_integration,
+        ).value = RegistrationConstants.SUCCESS_STATUS
```
Comment (tdilauro) [inline tests/test_controller.py:550]: Existing tests were expecting an active integration , so here we set the needed status.

File: tests/test_adobe_vendor_id.py [hunk 7]
```diff
@@ -624,6 +625,38 @@ def setup_method(self):
...
         )
 
+    @pytest.mark.parametrize(
+        "registration_status, authdata_utility_type",
+        [
+            (RegistrationConstants.SUCCESS_STATUS, AuthdataUtility),
+            (RegistrationConstants.FAILURE_STATUS, type(None)),
+            (None, type(None)),
+        ],
+    )
```
Comment (tdilauro) [inline tests/test_adobe_vendor_id.py:635]: New test to verify that we don't get an AuthdataUtility unless a successful registration has been accomplished.

**Review by jonathangreen — APPROVED** [review]
Looks good! 🎸🤘

**Pull request merged**